### PR TITLE
Poll change state after some actions

### PR DESCRIPTION
**Pull request by reiterl**
In  analog poll change the state if it created and votes are set in
poll.create, poll.update or option.update.

### CODE REVIEW

**Comment by reiterl** [comment]
Also remove type change from poll.update.

File: openslides_backend/action/actions/poll/update.py [hunk 1]
```diff
@@ -96,3 +96,19 @@ def check_100_percent_base(self, instance: Dict[str, Any]) -> None:
             )
             pollmethod = poll.get("pollmethod")
         base_check_100_percent_base(pollmethod, onehundred_percent_base)
+
+
+def check_state_change(instance: Dict[str, Any], poll: Dict[str, Any]) -> bool:
```
Comment (FinnStutzenstein) [inline openslides_backend/action/actions/poll/update.py:101]: Add this function as a member to the class.

File: openslides_backend/action/actions/option/update.py [hunk 2]
```diff
@@ -149,3 +169,14 @@ def _get_vote_id(
             if item["value"] == search_value:
                 return key
         return None
+
+
+def check_state_change(instance: Dict[str, Any], poll: Dict[str, Any]) -> bool:
+    if poll.get("type") != Poll.TYPE_ANALOG:
+        return False
+    if poll.get("state") != Poll.STATE_CREATED:
```
Comment (FinnStutzenstein) [inline openslides_backend/action/actions/option/update.py:177]: This check is not right. If a poll is *finished* and a user updates the poll and sets `publish_immediately`, the poll must be set to published.

Same for the other two functions.

**Review by FinnStutzenstein — CHANGES_REQUESTED** [review]

File: openslides_backend/action/actions/poll/create.py [hunk 3]
```diff
@@ -162,3 +164,22 @@ def check_100_percent_base(self, instance: Dict[str, Any]) -> None:
         pollmethod = instance["pollmethod"]
         onehundred_percent_base = instance.get("onehundred_percent_base")
         base_check_100_percent_base(pollmethod, onehundred_percent_base)
+
+    def check_state_change(self, instance: Dict[str, Any]) -> bool:
+        if instance["type"] != Poll.TYPE_ANALOG:
+            return False
+        check_fields = (
+            "amount_global_yes",
```
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/poll/create.py:172]: The `amount_global_*` logic removed from the `poll.create`.

File: openslides_backend/action/actions/option/update.py [hunk 4]
```diff
@@ -149,3 +169,14 @@ def _get_vote_id(
             if item["value"] == search_value:
                 return key
         return None
+
+
+def check_state_change(instance: Dict[str, Any], poll: Dict[str, Any]) -> bool:
```
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/option/update.py:174]: Done.

File: openslides_backend/action/actions/poll/update.py (hunk 1 above)
Comment (reiterl) [reply openslides_backend/action/actions/poll/update.py:101]: Done.

File: openslides_backend/action/actions/option/update.py (hunk 2 above)
Comment (reiterl) [reply openslides_backend/action/actions/option/update.py:177]: Update the publish_immediately logic.

Comment (FinnStutzenstein) [reply openslides_backend/action/actions/option/update.py]: This has to be done for all three methods.

**Comment by FinnStutzenstein** [comment]
One remark and rebasing is necessary.

File: openslides_backend/action/actions/option/update.py (hunk 2 above)
Comment (reiterl) [reply openslides_backend/action/actions/option/update.py]: What do you mean? I have updated that for all three actions.

File: openslides_backend/action/actions/option/update.py [hunk 5]
```diff
@@ -149,3 +179,15 @@ def _get_vote_id(
             if item["value"] == search_value:
                 return key
         return None
+
+    def check_state_change(
+        self, instance: Dict[str, Any], poll: Dict[str, Any]
+    ) -> bool:
+        if poll.get("type") != Poll.TYPE_ANALOG:
+            return False
+        if poll.get("state") != Poll.STATE_CREATED:
```
Comment (FinnStutzenstein) [inline openslides_backend/action/actions/option/update.py:188]: This check has to be removed (analog to poll.create)

File: openslides_backend/action/actions/poll/update.py [hunk 6]
```diff
@@ -96,3 +103,20 @@ def check_100_percent_base(self, instance: Dict[str, Any]) -> None:
             )
             pollmethod = poll.get("pollmethod")
         base_check_100_percent_base(pollmethod, onehundred_percent_base)
+
+    def check_state_change(
+        self, instance: Dict[str, Any], poll: Dict[str, Any]
+    ) -> bool:
+        if poll.get("type") != Poll.TYPE_ANALOG:
+            return False
+        if poll.get("state") != Poll.STATE_CREATED:
```
Comment (FinnStutzenstein) [inline openslides_backend/action/actions/poll/update.py:112]: This check has to be removed (analog to poll.create)

**Review by FinnStutzenstein — CHANGES_REQUESTED** [review]

File: openslides_backend/action/actions/option/update.py (hunk 5 above)
Comment (reiterl) [reply openslides_backend/action/actions/option/update.py:188]: For analog polls: If the state is created and at least one vote value is given (votesvalid/votesinvalid/votescast), the state must be set to finished.-> Is this wrong then?

File: openslides_backend/action/actions/poll/update.py (hunk 6 above)
Comment (reiterl) [reply openslides_backend/action/actions/poll/update.py:112]: see above.

File: openslides_backend/action/actions/option/update.py (hunk 5 above)
Comment (reiterl) [reply openslides_backend/action/actions/option/update.py:188]: https://github.com/OpenSlides/OpenSlides/wiki/poll.update

**Review by FinnStutzenstein — APPROVED** [review]
Ok! You were right about the created state.

**Comment by reiterl** [comment]
Rebased.

**Pull request merged**